packages/ember-views/lib/system/build-component-template.js

```diff
@@ -26,7 +26,8 @@ export default function buildComponentTemplate({ component, layout, isAngleBrack
     // element. We use `manualElement` to create a template that represents
     // the wrapping element and yields to the previous block.
     if (tagName !== '') {
-      var attributes = normalizeComponentAttributes(component, isAngleBracket, attrs);
+      var canChangeType = canChangeTypeAfterRender(attrs);
```

Comment:
This does not need to be done for every component rendered.  This can be done once in the root of the module and save the boolean value.  Changing that also allows you to avoid threading the boolean through the other invocations...